### PR TITLE
Authenticate requests that comes with backend JWT

### PR DESCRIPTION
**Pull request by shanakama**
## Purpose 
Fix : https://github.com/wso2-enterprise/choreo/issues/8542

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/BackendJWTAuthenticationImpl.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                String jwtTokenIdentifier = getJWTTokenIdentifier(signedJWTInfo);
+
+                JWTValidationInfo jwtValidationInfo = new JWTValidationInfo();
+                jwtValidationInfo.setValid(true);
+                if (isRESTApiTokenCacheEnabled) {
+                    getRESTAPITokenCache().put(jwtTokenIdentifier, jwtValidationInfo);
+                }
+                //Validating scopes
+                OauthUtils oauthUtils = new OauthUtils();
+                return oauthUtils.handleScopeValidation(message, signedJWTInfo, token);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/BackendJWTAuthenticationImpl.java:77]: Lets change the class name and make this a static method.
```suggestion
                return JWTUtil.handleScopeValidation(message, signedJWTInfo, token);
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java [hunk 2]
```diff
@@ -71,14 +73,21 @@ public void handleMessage(Message inMessage) {
         //set the request_authentication_scheme property in the message as oauth2.
         String accessToken = RestApiUtil.extractOAuthAccessTokenFromMessage(inMessage,
                 RestApiConstants.REGEX_BEARER_PATTERN, RestApiConstants.AUTH_HEADER_NAME);
+        String backendJWT = RestApiUtil.extractOAuthAccessTokenFromMessage(inMessage,
+                RestApiConstants.REGEX_BEARER_PATTERN, RestApiConstants.BACKEND_JWT_AUTH_HEADER_NAME);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java:77]: Should the backend JWT come in x-jwt-assertion: Bearer <token> format?
It is coming via just x-jwt-assertion: <token>

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java [hunk 3]
```diff
@@ -71,14 +73,21 @@ public void handleMessage(Message inMessage) {
...
-        if (accessToken.contains(RestApiConstants.DOT)) {
+        if ((accessToken != null && accessToken.contains(RestApiConstants.DOT))) {
             inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.JWT_AUTHENTICATION);
+            inMessage.put(RestApiConstants.JWT_TOKEN, accessToken);
+        } else if ((backendJWT != null && backendJWT.contains(RestApiConstants.DOT))) {
+            inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.BACKEND_JWT_AUTHENTICATION);
+            inMessage.put(RestApiConstants.JWT_TOKEN, backendJWT);
         } else {
             inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.OPAQUE_AUTHENTICATION);
         }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/auth/OAuthAuthenticationInterceptor.java:93]:
```suggestion
        if ((accessToken != null && accessToken.contains(RestApiConstants.DOT))) {
            inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.JWT_AUTHENTICATION);
            inMessage.put(RestApiConstants.JWT_TOKEN, accessToken);
            inMessage.put(RestApiConstants.MASKED_TOKEN, APIUtil.getMaskedToken(accessToken));
        } else if ((backendJWT != null && backendJWT.contains(RestApiConstants.DOT))) {
            inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.BACKEND_JWT_AUTHENTICATION);
            inMessage.put(RestApiConstants.JWT_TOKEN, backendJWT);
            inMessage.put(RestApiConstants.MASKED_TOKEN, APIUtil.getMaskedToken(backendJWT));
        } else {
            inMessage.put(RestApiConstants.REQUEST_AUTHENTICATION_SCHEME, RestApiConstants.OPAQUE_AUTHENTICATION);
            inMessage.put(RestApiConstants.MASKED_TOKEN, APIUtil.getMaskedToken(accessToken));
        }
```

**Comment by shanakama** [comment]
Closing this PR since these changes should be moved to another repository.

**Pull request closed without merging**